Add many stats to single-node graphs page

cuongdo opened the pull request:
Resolves #5275

I had to restructure NodePage slightly. I pulled out the graph
construction into a NodePage._initGraphs(), so that it can be called
multiple times. We need this, because we need the store_ids from our
NodeStatus, which is not available when the constructor is called.

Also made rates per second instead of per sample period.

Currently, when retrieving derivatives from the TS DB, we return the rate
per sample period. It seems more generally useful to return rate per
second, for metrics like queries per second and bytes per second.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5395)

<!-- Reviewable:end -->

Commit:
Make rates per second instead of per sample period

Currently, when retrieving derivates from the TS DB, we return the rate
per sample period. It seems more generally useful to return rate per
second, for metrics like queries per second and bytes per second.

Commit:
Add most stats to single-node graphs page

I had to restructure NodePage slightly. I pulled out the graph
construction into a NodePage._initGraphs(), so that it can be called
multiple times. We need this, because we need the store_ids from our
NodeStatus, which is not available when the constructor is called.

Resolves #5275

cuongdo commented:
<img width="1411" alt="screenshot 2016-03-18 22 41 30" src="https://cloud.githubusercontent.com/assets/1564582/13896239/358610fe-ed5b-11e5-93ae-185726f7e651.png">
<img width="1372" alt="screenshot 2016-03-18 22 41 57" src="https://cloud.githubusercontent.com/assets/1564582/13896240/3b300816-ed5b-11e5-8fc7-305faeaaea9e.png">

petermattis commented:
Very nice to see these additional graphs.

I thought you determined that we would never see memtable hits and misses because we're using iterators and not "Get" operations.

Any idea why the clock-offset upper and lower bounds are pegged at +/- 250ms? That doesn't seem correct.

An additional system graph I'd like to see is GC pause time. I'm pretty sure we're already collecting this.

Agreed that it makes more sense for the graphs to display per sec rates instead of per sampling period.

tamird commented on ts/query.go:
```diff
@@ -535,7 +536,7 @@ func (db *DB) Query(query Query, r Resolution, startNanos, endNanos int64) ([]*T
 		*response = current
 		if isDerivative {
-			dTime := (current.TimestampNanos - last.TimestampNanos) / r.SampleDuration()
+			dTime := (current.TimestampNanos - last.TimestampNanos) / int64(time.Second)
```
nit: https://godoc.org/time#Duration.Nanoseconds

tamird commented:
Reviewed 3 of 3 files at r1, 2 of 3 files at r2.
Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5395)*
<!-- Sent from Reviewable.io -->

cuongdo commented:
I'm not sure what's going on with the clock offset. I'm just recording what I see. I added a `Println` to `(*RemoteClockMonitor).MonitorRemoteOffsets` ([here](https://gist.github.com/cuongdo/21eb5d499a10a6342198)), and I see this `offsetInterval` being reported for a local three-node cluster:

```
[-250.090711ms, 250.088507ms]
[-250.104969ms, 250.096665ms]
[-250.079898ms, 250.105435ms]
[-250.087016ms, 250.07888ms]
[-250.08287ms, 250.095749ms]
[-250.0814ms, 250.114184ms]
[-250.074172ms, 250.131096ms]
[-250.067711ms, 250.109943ms]
[-250.07151ms, 250.109635ms]
[-250.092365ms, 250.126648ms]
[-250.080465ms, 250.202905ms]
[-250.07304ms, 250.093187ms]
[-250.091783ms, 250.124188ms]
[-250.107987ms, 250.088472ms]
```

I've added a GC graph and removed memtable hits/misses:

![Screenshot 2016-03-21 11.42.27.png](https://files.reviewableusercontent.io/images/reviewable/1564582/dFdjC4l100Y90wzCr7MyhqVe/Screenshot%202016-03-21%2011.42.27.png)

---

Review status: 2 of 6 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[ts/query.go, line 539 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5395#-KDBkasrdba2xRmZayHD-r2-539:-KDOfm8HNaInkrpARw2W:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/cccf70a081e74c972943e1eb465b49cb76d3fe95/ts/query.go#L539)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5395#-:-KDOgy3chhnM6KDqDOn-:-1618229196)*
<!-- Sent from Reviewable.io -->

mrtracy commented:
LGTM; thanks for fixing the rate denominator.
